sdk/cosmos/azure-cosmos-spark_3-0_2-12/docs/configuration-reference.md

```diff
@@ -0,0 +1,59 @@
+Configuration Reference:
+
+
+## Generic Configuration
+
+
+| Config Property Name      | Default | Description |
+| :---        |    :----   |         :--- | 
+| `spark.cosmos.accountEndpoint`      | None   | Cosmos DB Account Endpoint Uri |
+| `spark.cosmos.accountKey`      | None    | Cosmos DB Account Key  |
+| `spark.cosmos.database`      | None    | Cosmos DB database name  |
+| `spark.cosmos.container`      | None    | Cosmos DB container name  |
+
+
+### Additional tuning
+
+
+
+| Config Property Name      | Default | Description |
+| :---        |    :----   |         :--- | 
+| `spark.cosmos.useGatewayMode`      | `false`    | Use gateway mode for the client operations  |
+| `spark.cosmos.read.forceEventualConsistency`      | `true`    | Makes the client use Eventual consistency for read operations  |
+| `spark.cosmos.applicationName`      | None    | Application name  |
+
+
+### Write Config
+
+
+| Config Property Name      | Default | Description |
+| :---        |    :----   |         :--- | 
+| `spark.cosmos.write.strategy`      | `ItemOverwrite`    | Cosmos DB Item write Strategy: ItemOverwrite (using upsert), ItemAppend (using create, ignore 409)  |
+| `spark.cosmos.write.maxRetryCount`      | `3`    | Cosmos DB Write Max Retry Attempts on failure  |
+| `spark.cosmos.write.bulkEnabled`      | `false`   | Cosmos DB Item Write bulk enabled |
```

Comment:
Should default to true

Reply:
will change to true after I add the semaphore.